mods/variations/init.lua

```diff
@@ -0,0 +1,105 @@
+variations = {}
+
+variations.variations = {
+	{
+		name = "brick";
+		description = "Brick";
+		texture = "0,0";
+	},
+	{
+		name = "big_tile";
+		description = "Big Tile";
+		texture = "1,0";
+	},
+	{
+		name = "tile";
+		description = "Tile";
+		texture = "2,0";
+	},
+	{
+		name = "tall_tile";
+		description = "Tall Tile";
+		texture = "0,1";
+	},
+	{
+		name = "small_brick";
+		description = "Small Brick";
+		texture = "1,1";
+	},
+	{
+		name = "small_tile";
+		description = "Small Tile";
+		texture = "2,1";
+	},
+	{
+		name = "horizontal_brick";
+		description = "Horizontal Brick";
+		texture = "0,2";
+	},
+	{
+		name = "cross_tile";
+		description = "Cross Tile";
+		texture = "1,2";
+	},
+	{
+		name = "spiral_tile";
+		description = "Spiral Tile";
+		texture = "2,2";
+	},
+}
+
+function variations.register_for_base(base_node)
+	local base_definition = minetest.registered_nodes[base_node]
+	for _, variation in ipairs(variations.variations) do
+		local sunlight = true
+		-- We get an iterator function over substrings split by :
+		local base_node_full_name = string.gmatch(base_node, "([^:]+)")
+		-- First call will give us the modname
+		-- local base_node_modname = base_node_full_name()
+		-- Unused variable warning...
+		base_node_full_name()
+		-- and the second will give us the node name
+		local base_node_name = base_node_full_name()
+        local transparency = base_node_name.use_texture_alpha
+        if transparency == nil then
+			sunlight = false
+            transparency = "opaque"
```

Comment:
this won't be required, if not defined, transparency will be opaque by default anyways